agent/engine/serviceconnect/manager_linux.go

```diff
@@ -200,6 +202,37 @@ func defaultMkdirAllAndChown(path string, perm fs.FileMode, uid, gid int) error
 	return nil
 }
 
+func getRegionFromContainerInstanceARN(containerInstanceARN string) string {
+	// Parse the ARN
+	parsedARN, err := arn.Parse(containerInstanceARN)
+	if err != nil {
+		return ""
+	}
+
+	// Extract the region from the parsed ARN
+	return parsedARN.Region
+}
+
+func isIsoRegion(region string) bool {
+	partition, ok := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region)
+	if !ok {
+		// if partition is not found, assume it's iso
```

Comment:
Can we explicitly check if partition has prefix "aws-iso" for determining if a partition is ISO?

Proposing to check this way because in the future if a new non ISO partition is added, then it will inadvertently be brought into the blast radius until this function is updated with the new partition.

Reply:
Discussed offline with Cam. We aligned that the risk for potentially including new non-ISO regions as in scope for receiving the `/etc/pki mount` is minimal and not concerning (and far outweighs the risk of excluding those regions but inadvertently failing to include ISO regions as in scope for receiving the `/etc/pki mount`).